app/controllers/account/pages_controller.rb

```diff
@@ -7,13 +7,26 @@ def show
     @page = resource
   end
 
+  def rate
+    @page = resource
+    points = params[:commit]
+    last_rating = @page.rating
+    if Page::LEGAL_RATING.include?(points)
+      @page.update_attribute(:rating, last_rating + points.to_i)
+      redirect_to  account_company_pages_path
```

Comment:
A lot of logic here should be moved to model.
should look like:
```ruby
if @page.rate(params[:commit])
  redirect_to  account_company_pages_path
else
...
```
Also you should avoid using update_attribute: http://www.davidverhasselt.com/set-attributes-in-activerecord/